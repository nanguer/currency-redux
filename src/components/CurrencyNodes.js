import React from "react";
import Currency from "../components/Currency";

const CurrencyNodes = ({ currencies }) => (
  <>
    {currencies.length > 0 ? (
      currencies.map((currency, i) => {
        return <Currency key={i} currency={currency} />;
      })
    ) : (
      <p>No currencies...</p>
    )}
  </>
);

export default CurrencyNodes;
