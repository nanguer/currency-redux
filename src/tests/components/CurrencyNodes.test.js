import React from "react";
import { shallow } from "enzyme";
import rates from "../fixtures/rates";
import CurrencyNodes from "../../components/CurrencyNodes";

test("Should render CurrencyNodes with currency", () => {
  const wrapper = shallow(<CurrencyNodes currencies={rates} />);
  expect(wrapper).toMatchSnapshot();
});

test("Should render CurrencyNodes with no currencies message", () => {
  const wrapper = shallow(<CurrencyNodes currencies={[]} />);
  expect(wrapper).toMatchSnapshot();
});
