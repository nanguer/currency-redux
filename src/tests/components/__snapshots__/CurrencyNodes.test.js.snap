// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CurrencyNodes with currency 1`] = `
<Fragment>
  <Connect(ConnectedCurrency)
    currency={
      Object {
        "code": "CZK",
        "currency": "korona czeska",
        "mid": 0.1698,
      }
    }
    key="0"
  />
  <Connect(ConnectedCurrency)
    currency={
      Object {
        "code": "DKK",
        "currency": "korona duÅ„ska",
        "mid": 0.5729,
      }
    }
    key="1"
  />
  <Connect(ConnectedCurrency)
    currency={
      Object {
        "code": "ISK",
        "currency": "korona islandzka",
        "mid": 0.031457,
      }
    }
    key="2"
  />
</Fragment>
`;

exports[`Should render CurrencyNodes with no currencies message 1`] = `
<Fragment>
  <p>
    No currencies...
  </p>
</Fragment>
`;
